command/validate.go

```diff
@@ -64,48 +61,38 @@ func (c *ValidateCommand) RunContext(ctx context.Context, cla *ValidateArgs) int
 		return 0
 	}
 
-	// Get the core
-	core, err := c.Meta.Core(tpl, &cla.MetaArgs)
-	if err != nil {
-		c.Ui.Error(err.Error())
-		return 1
-	}
-
 	errs := make([]error, 0)
 	warnings := make(map[string][]string)
 
-	// Get the builds we care about
-	buildNames := core.BuildNames(c.CoreConfig.Only, c.CoreConfig.Except)
-	builds := make([]packer.Build, 0, len(buildNames))
-	for _, n := range buildNames {
-		b, err := core.Build(n)
-		if err != nil {
-			c.Ui.Error(fmt.Sprintf(
-				"Failed to initialize build '%s': %s",
-				n, err))
-			return 1
-		}
+	_, diags := packerStarter.GetBuilds(packer.GetBuildsOptions{
+		Only:   cla.Only,
+		Except: cla.Except,
+	})
 
-		builds = append(builds, b)
+	// here, something could have gone wrong but we still want to run valid
+	if ret := writeDiags(c.Ui, nil, diags); ret != 0 {
+		return ret
 	}
 
-	// Check the configuration of all builds
-	for _, b := range builds {
-		log.Printf("Preparing build: %s", b.Name())
-		warns, err := b.Prepare()
-		if len(warns) > 0 {
-			warnings[b.Name()] = warns
-		}
-		if err != nil {
-			errs = append(errs, fmt.Errorf("Errors validating build '%s'. %s", b.Name(), err))
-		}
+	if cfgType, _ := ConfigType(cla.Path); cfgType == "hcl" {
+		c.Ui.Say("Template validated successfully.")
+		return 0
+	}
```

Comment:
Oh smart that is a good & simple way to make `validate` work, I had in mind to change the `packerStarter` interface and give it a `Validate` function so that the structs know how to validate themselves but I think this is also good 🙂

Reply:
Nice, I like the idea of updating the interface. I think the validate command knows too much about what is going on behind the scenes :laughing: So I think the refactor is in order. 

But it is good to see that we have a working solution to test against. Thanks for looking over the changes.